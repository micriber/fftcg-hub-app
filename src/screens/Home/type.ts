export type HomeStackParamList = {
  CollectionSearch: undefined;
  Home: undefined;
};
