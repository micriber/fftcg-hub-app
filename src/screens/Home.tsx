import React from 'react';
import {StackNavigationProp} from '@react-navigation/stack';
import {
  View,
  Image,
  StyleSheet,
  FlatList,
  Dimensions,
  Text,
  TextInput,
} from 'react-native';
import {useAsync} from 'react-async';
import {AuthContext} from '../AuthContext';
import Loading from './Loading';
import cards from '../../jest/mocks/fftcg';
import FastImage from 'react-native-fast-image';

export type RootStackParamList = {
  Home: undefined;
  About: undefined;
  Login: undefined;
};

type ProfileScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'Home'
>;

type Props = {
  navigation: ProfileScreenNavigationProp;
};

const w = Dimensions.get('window');

const Home = () => {
  const {signOut, getCurrentUser} = React.useContext(AuthContext);

  const {data, isLoading} = useAsync({
    promiseFn: getCurrentUser,
  });

  if (isLoading) {
    return <Loading />;
  }

  if (!data) {
    signOut();
    return null;
  }

  return (
    <View style={[styles.container]}>
      <TextInput
        blurOnSubmit
        autoCapitalize="none"
        placeholder={'Rechercher "1-001R"'}
        autoCorrect={false}
        style={{
          borderRadius: 25,
          borderColor: '#333',
          backgroundColor: '#fff',
          paddingLeft: 50,
        }}
        // textStyle={{color: '#000'}}
      />
      <View style={[styles.cardContainer]}>
        <FlatList
          initialNumToRender={2}
          numColumns={3}
          data={cards}
          keyExtractor={(item) => item.Code}
          renderItem={({item}) => (
            <View style={{flex: 1, flexDirection: 'column', margin: 1}}>
              <FastImage
                // style={styles.image}
                style={{width: w.width / 4, height: w.height / 4}}
                source={{
                  uri: `https://fftcg.cdn.sewest.net/images/cards/thumbs/${item.Code}_fr.jpg`,
                }}
                resizeMode="contain"
              />
            </View>
          )}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 30,
    marginLeft: 20,
    marginRight: 20,
  },
  cardContainer: {
    // flex: 1,
    marginTop: 3,
    marginLeft: 15,
    // flexDirection: 'row',
    // alignItems: 'flex-start',
    // justifyContent: 'space-between',
  },
  // image: {
  //   height: '25%',
  //   width: '25%',
  // },
});

export default Home;
