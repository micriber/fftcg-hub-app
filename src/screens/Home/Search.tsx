import React from 'react';
import {View, Button, TextInput, StyleSheet} from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';
import {HomeStackParamList} from './type';

type SearchScreenNavigationProp = StackNavigationProp<
  HomeStackParamList,
  'CollectionSearch'
>;

type Props = {
  navigation: SearchScreenNavigationProp;
};

const Search = ({navigation}: Props) => {
  return (
    <View style={styles.container}>
      <TextInput
        blurOnSubmit
        autoCapitalize="none"
        placeholder={'Rechercher "1-001R"'}
        autoCorrect={false}
        style={[styles.searchInput]}
      />
      <Button title="Submit" onPress={() => navigation.navigate('Home')} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 30,
    marginLeft: 20,
    marginRight: 20,
  },
  searchInput: {
    borderRadius: 25,
    borderColor: '#333',
    backgroundColor: '#fff',
    paddingLeft: 50,
  },
});

export default Search;
