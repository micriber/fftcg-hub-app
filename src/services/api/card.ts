import {config} from '../../config';

export type Card = {
  id: string;
  code: string;
  element: string;
  rarity: string;
  cost: string;
  power: string;
  category1: string;
  category2: string;
  multicard: string;
  exBurst: string;
  name: string;
  type: string;
  job: string;
  text: string;
};

export type Cards = Card[];
export type GetCardsResponse = {
  cards: Cards;
  total: number;
};

export type UnauthorizedError = {
  message: string[];
};

export const getCards = async (
  token: string,
): Promise<GetCardsResponse | UnauthorizedError> => {
  return fetch(`${config.api.baseUri}/api/v1/cards`, {
    method: 'GET',
    headers: {
      authorization: `bearer ${token}`,
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
  }).then((response) => response.json());
};
