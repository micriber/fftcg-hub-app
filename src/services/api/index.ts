interface IApiConfigure {
  token?: string;
  refreshCallback?: () => {};
}

interface IRefreshWrapperOptions {
  json?: boolean;
}

export class Api {
  private static instance: Api;
  private token?: string;
  private refreshCallback?: () => {};

  static set Token(newToken: string | undefined) {
    const api = Api.getInstance();
    api.token = newToken;
  }

  /**
   * The Singleton's constructor should always be private to prevent direct
   * construction calls with the `new` operator.
   */
  private constructor() {}

  public static getInstance(): Api {
    if (!Api.instance) {
      Api.instance = new Api();
    }

    return Api.instance;
  }

  public static configure({token, refreshCallback}: IApiConfigure) {
    const api = Api.getInstance();

    if (token) {
      api.token = token;
    }

    if (refreshCallback) {
      api.refreshCallback = refreshCallback;
    }
  }

  private renderResponse(response: Response, json: boolean) {
    if (json) {
      return response.json();
    }

    return response;
  }

  public async refreshWrapper(
    callback: (token: string) => Promise<Response>,
    {json = true}: IRefreshWrapperOptions,
  ) {
    const res = await callback(this.token!);

    if (res.status === 401) {
      this.refreshCallback && (await this.refreshCallback());
      const resAfterRefresh = await callback(this.token!);

      return this.renderResponse(resAfterRefresh, json);
    }

    return this.renderResponse(res, json);
  }
}
