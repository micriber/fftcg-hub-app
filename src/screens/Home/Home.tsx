import React from 'react';
import {Alert, Text} from 'react-native';
import useAsync from '../../utils/hooks/useAsync';
import {getCards} from '../../services/api/card';
import {AuthContext} from '../../AuthContext';
import Loading from '../Loading';
import FFCardsList from '../../components/FFCardsList';
import HeaderSwitch from '../../components/common/HeaderSwitch';
import BottomRightButton from '../../components/common/BottomRightButton';

const Home = () => {
  const [isListView, setIsListView] = React.useState(false);
  const {getIdToken} = React.useContext(AuthContext);
  const token = getIdToken();
  const state = useAsync(async () => {
    return await getCards({token: token!, params: {}});
  }, []);

  if (state.loading) {
    return <Loading />;
  }
  if (state.error) {
    return <Text>{JSON.stringify(state.error)}</Text>;
  }
  if (!state.value || (state.value && 'message' in state.value)) {
    return <Text>{JSON.stringify(state.value?.message)}</Text>;
  }

  const cards = state.value.data;

  return (
    <>
      <HeaderSwitch
        leftIconName="appstore-o"
        rightIconName="bars"
        value={isListView}
        onValueChange={setIsListView}
      />
      <FFCardsList isListView={isListView} cards={cards} />
      <BottomRightButton
        iconName="search1"
        onPress={() => Alert.alert('Search button pressed')}
      />
    </>
  );
};

export default Home;
