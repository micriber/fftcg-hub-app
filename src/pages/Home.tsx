import React from 'react';
import {StackNavigationProp} from '@react-navigation/stack';
import {Button, SafeAreaView, ActivityIndicator, Text} from 'react-native';
import {useAsync} from 'react-async';
import {GoogleSignin} from '@react-native-community/google-signin';
import {signOut} from '../services/google';

export type RootStackParamList = {
  Home: undefined;
  About: undefined;
  Login: undefined;
};

type ProfileScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'Home'
>;

type Props = {
  navigation: ProfileScreenNavigationProp;
};

const Home = ({navigation}: Props) => {
  const logoutAndRedirect = async () => {
    await signOut();
    navigation.navigate('Login');
  };
  const {data, error, isLoading} = useAsync({
    promiseFn: GoogleSignin.getCurrentUser,
  });
  if (isLoading) {
    return <ActivityIndicator size="small" color="#0000ff" />;
  }
  if (error) {
    return `Something went wrong: ${error.message}`;
  }

  return (
    <SafeAreaView>
      <Text>Welcome {data?.user.name}</Text>
      <Button
        title="Go to about page"
        onPress={() => navigation.navigate('About')}
      />
      <Button title="Logout" onPress={logoutAndRedirect} />
    </SafeAreaView>
  );
};

export default Home;
