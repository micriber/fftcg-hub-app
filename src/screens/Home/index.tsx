import React from 'react';
import {createStackNavigator, StackHeaderProps} from '@react-navigation/stack';
import Home from './Home';
import Search from './Search';
import CardDetails from '../Cards/CardDetail';
import Header from '../../components/navigation/header';

const HomeStack = createStackNavigator();

const HomeStackScreen = () => (
  <HomeStack.Navigator
    screenOptions={{
      header: (headerProps: StackHeaderProps) => <Header {...headerProps} />,
    }}>
    <HomeStack.Screen
      options={{
        headerTitle: 'Home',
        headerShown: true,
      }}
      name="Home"
      component={Home}
    />
    <HomeStack.Screen name="CollectionSearch" component={Search} />
    <HomeStack.Screen
      name="CardDetails"
      component={CardDetails}
      // TODO: Remove this any
      options={(props) => {
        const {route} = props;
        const pageTitle = route.params && (route.params as any).pageTitle;
        return {
          ...(pageTitle ? {title: pageTitle} : {}),
          headerShown: true,
        };
      }}
    />
  </HomeStack.Navigator>
);

export {HomeStack, HomeStackScreen};
