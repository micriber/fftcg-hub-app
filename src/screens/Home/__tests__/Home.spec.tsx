/**
 * @format
 */

import 'react-native';
import React from 'react';
import Home from '../Home';
import {
  fireEvent,
  render,
  waitForElementToBeRemoved,
} from '@testing-library/react-native';
import {Alert} from 'react-native';
jest.mock('../../../services/api/card', () => ({
  getCards: jest.fn(),
}));
import {getCards} from '../../../services/api/card';

const data = [
  {
    code: '1-001H',
    name: 'Boulou',
    text: 'billy',
    type: 'fire',
    element: 'blue',
  },
  {
    code: '1-002R',
    name: 'Boulou',
    text: 'billy',
    type: 'fire',
    element: 'blue',
  },
  {
    code: '1-003C',
    name: 'Boulou',
    text: 'billy',
    type: 'fire',
    element: 'blue',
  },
];

// More about testing in this repository:
// https://github.com/bamlab/react-native-testing/blob/master/src/pages/Home/__tests__/Home.test.tsx
describe('[Page] Home', () => {
  it('renders correctly', async () => {
    (getCards as jest.Mock).mockResolvedValue({data});
    const root = render(<Home />);

    expect(root.toJSON()).toMatchSnapshot(); // Loading screen
    await waitForElementToBeRemoved(() => root.getByTestId('Loader'));
    expect(root.toJSON()).toMatchSnapshot(); // Home page loaded with cards
  });

  it('should show error', async () => {
    (getCards as jest.Mock).mockRejectedValue({
      message: 'INTERNAL_SERVER_ERROR',
    });

    const root = render(<Home />);

    expect(root.toJSON()).toMatchSnapshot(); // Loading screen
    await waitForElementToBeRemoved(() => root.getByTestId('Loader'));
    expect(root.toJSON()).toMatchSnapshot(); // Home page loaded with cards
  });

  it('should show a backend error message', async () => {
    (getCards as jest.Mock).mockResolvedValue({
      message: '401 Unauthorized',
    });

    const root = render(<Home />);

    expect(root.toJSON()).toMatchSnapshot(); // Loading screen
    await waitForElementToBeRemoved(() => root.getByTestId('Loader'));
    expect(root.toJSON()).toMatchSnapshot(); // Home page loaded with cards
  });

  it('should show an alert on press', async () => {
    (getCards as jest.Mock).mockResolvedValue({data});
    Alert.alert = jest.fn();

    const root = render(<Home />);

    // expect(root.toJSON()).toMatchSnapshot(); // Loading screen
    await waitForElementToBeRemoved(() => root.getByTestId('Loader'));
    fireEvent(root.getByTestId('BottomRightButton'), 'onPress');
    expect(Alert.alert).toHaveBeenCalledTimes(1);
    // expect(root.toJSON()).toMatchSnapshot(); // Home page loaded with cards
  });
});
