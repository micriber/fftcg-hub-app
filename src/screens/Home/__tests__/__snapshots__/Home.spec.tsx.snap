// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page] Home renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#878683",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="SwitcherState-0"
      value={false}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="Loader"
  >
    <ActivityIndicator
      animating={true}
      color="#0000ff"
      hidesWhenStopped={true}
      size="large"
    />
  </View>,
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 20,
        "height": 75,
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00BCD4",
          "borderColor": "#fff",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="BottomRightButton"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`[Page] Home renders correctly 2`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#878683",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="SwitcherState-0"
      value={false}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "flexGrow": 1,
          "justifyContent": "flex-start",
          "width": "100%",
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <Text
          style={
            Object {
              "marginTop": "50%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          No result
        </Text>
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={2}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={1}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <Text
          onLayout={[Function]}
          style={
            Object {
              "marginTop": "50%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          No result
        </Text>
      </View>
    </RCTScrollView>
  </View>,
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 20,
        "height": 75,
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00BCD4",
          "borderColor": "#fff",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="BottomRightButton"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`[Page] Home should show a backend error message 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#878683",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="SwitcherState-0"
      value={false}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="Loader"
  >
    <ActivityIndicator
      animating={true}
      color="#0000ff"
      hidesWhenStopped={true}
      size="large"
    />
  </View>,
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 20,
        "height": 75,
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00BCD4",
          "borderColor": "#fff",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="BottomRightButton"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`[Page] Home should show a backend error message 2`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#878683",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="SwitcherState-0"
      value={false}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>,
  <Text>
    "401 Unauthorized"
  </Text>,
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 20,
        "height": 75,
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00BCD4",
          "borderColor": "#fff",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="BottomRightButton"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`[Page] Home should show error 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#878683",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      testID="SwitcherState-0"
      value={false}
    />
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="Loader"
  >
    <ActivityIndicator
      animating={true}
      color="#0000ff"
      hidesWhenStopped={true}
      size="large"
    />
  </View>,
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "bottom": 20,
        "height": 75,
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00BCD4",
          "borderColor": "#fff",
          "borderRadius": 50,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="BottomRightButton"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`[Page] Home should show error 2`] = `null`;
